packages/react-component-library/src/components/Searchbar/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef } from 'react'
+import { Form, Field, Formik } from 'formik'
+
+import withFormik from '../../enhancers/withFormik'
+import TextInput from '../TextInput'
+import { RightArrow } from '../../icons'
+
+const FormikTextInput = withFormik(TextInput)
+
+interface SearchFormType {
+  term: string
+}
+
+const initialValues: SearchFormType = { term: '' }
+
+export interface SearchbarProps {
+  className?: string
+  onSearch: (term: string) => void
+  searchButton: any
+  searchPlaceholder: string
+  setShowSearch: (visible: boolean) => void
+  style?: object
+}
+
+export const Searchbar: React.FC<SearchbarProps> = ({
+  className = '',
+  onSearch,
+  searchButton,
+  searchPlaceholder,
+  setShowSearch,
+  ...rest
+}) => {
+  const searchBoxRef = useRef()
+
+  const onSubmit = ({ term }: SearchFormType) => {
+    onSearch(term)
+  }
+
+  function documentClick(event: Event) {
+    // workaround for undefined error in typescript
+    const current = searchBoxRef.current || {
+      contains: (target: any): boolean => target === null,
+    }
+
+    if (
+      current.contains(event.target) ||
+      searchButton.current.contains(event.target)
+    ) {
+      return
+    }
+
+    setShowSearch(false)
+  }
+
+  useEffect(() => {
+    document.addEventListener('mousedown', documentClick)
+
+    return () => {
+      document.removeEventListener('mousedown', documentClick)
+    }
+  }, [])
+
+  return (
+    <Formik initialValues={initialValues} onSubmit={onSubmit}>
```

Comment:
It is being tightly coupled to Formik here. Could we avoid this maybe?

Reply:
its coupled internally so the outside world is not aware, or care. I thought it would be ok. All that the consumer is bothered about is that they get a call to a function with the term to search for

Reply:
Though I may have a look and see if I could strip Formik out as it's a single field